staging/src/k8s.io/client-go/discovery/discovery_client.go

```diff
@@ -375,19 +375,18 @@ func (d *DiscoveryClient) SwaggerSchema(version schema.GroupVersion) (*swagger.A
 	return &schema, nil
 }
 
-// OpenAPISchema fetches the open api schema using a rest client and parses the json.
-// Warning: this is very expensive (~1.2s)
-func (d *DiscoveryClient) OpenAPISchema() (*spec.Swagger, error) {
-	data, err := d.restClient.Get().AbsPath("/swagger.json").Do().Raw()
+// OpenAPISchema fetches the open api schema using a rest client and parses the proto.
+func (d *DiscoveryClient) OpenAPISchema() (*openapi_v2.Document, error) {
+	data, err := d.restClient.Get().AbsPath("/swagger-2.0.0.pb-v1").Do().Raw()
```

Comment:
> It shouldn't. 1.7 server already supports that end-point, and this code is meant for 1.8 client.

I rather not have this be the thing that breaks generic commands (like create) from working.  If it returns an error, will kubectl still work?  If not, at least guarding it sufficiently to keep that working seems reasonable even if client side validation doesn't work.